### ISSUE TITLE
Controls dont work after some time idle

### ISSUE THREAD

**Issue by Oxize** [dvx/lofi#31]
After a while when it played music, or when its paused, the controls dont work anymore. You can click whatever you want but not works.

When you go back to the main spotify player and press play again, the widget reacts again for a while.

**Comment by dvx** [comment]
It might be due to a lost internet connection (or if you disconnect/reconnect WiFi). The controls' state depends on Spotify's API responses.

**Comment by janjur** [comment]
I have the same issue. I am not able to control spotify after some idle time. Also, after similar amount of time wireless headphones control stops working. Both cases may have something to do with apparent Spotify's feature of blocking API after some idle time.
Maybe consider turning volume up and down to previous level every 5 seconds to keep it active?

**Comment by dvx** [comment]
I didn't realize this was a Spotify limitation @janjur -- will definitely implement a "heartbeat" if this is the case, although I'm already polling every 1 second (for play state, even if paused), so it technically should never lose connection with the Spotify server.

**Comment by dvx** [comment]
I was able to reproduce this today -- need to figure out a way to fix it, thanks for the report!

**Comment by dvx** [comment]
After a while, if not playing, it looks like Spotify clears the `current_device` so then all calls return:

```
{
  "error" : {
    "status" : 404,
    "message" : "Player command failed: No active device found",
    "reason" : "NO_ACTIVE_DEVICE"
  }
}
```
Maybe a play + pause or a volume jiggle will make the current device "sticky" so Spotify doesn't reset it.

**Comment by janjur** [comment]
If paused:
- volume to 0
- play for 0.1 seconds
- go back 0.1 seconds
- volume to previous value
--- repeat every minute :D

**Comment by dvx** [comment]
Fixed by #33. Thanks for the report. I was able to fix it by simply making a `https://api.spotify.com/v1/me/player/seek` request to the current scrubber location when the player is paused. This acts as a "keep alive."

**Comment by janjur** [comment]
Works flawlessly now :) I can go back from kitchen after 15 mins of idle, double tap the headset, it starts playing at once. Nice job, thank you!
